public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -378,6 +399,60 @@ export class TempoDatasource extends DataSourceWithBackend<TempoQuery, TempoJson
     };
   }
 
+  handleMetricsSummary = (target: TempoQuery, query: string, options: DataQueryRequest<TempoQuery>) => {
+    const groupBy = target.groupBy
+      ?.filter((f) => f.tag)
+      .map((f) => {
+        if (f.scope === TraceqlSearchScope.Unscoped) {
+          return `.${f.tag}`;
+        }
+        return f.scope !== TraceqlSearchScope.Intrinsic ? `${f.scope}.${f.tag}` : f.tag;
+      })
+      .join(', ');
+
+    return this._request('/api/metrics/summary', {
+      q: query,
+      groupBy,
+      start: options.range.from.unix(),
+      end: options.range.to.unix(),
+    }).pipe(
+      startWith({
```

Comment:
Why do we start with this? Isn't loading state done automatically as query is started?

Reply:
The query is sent off and returns nothing until it has all the data. Perhaps it's not supposed to work like that? This is so the user knows that the request was sent and is in loading state.

Reply:
I tried some changes cause I wasn't exactly sure but you can see them here https://github.com/grafana/grafana/pull/73666 and it seems like it works fine. It even get rid a short flicker when it seems we would show `0 series retrieved` which I assume is because of that one loading frame returned. I think loading state for simple requests, which just return single response (so no streaming, multiresponse or pooling) should be handled by explore/dashboard and you don't need to do anything special there.

Reply:
Decided it's prob better to not add custom loading state logic for just this use case to the UI.